R/toml_serialization.R

```diff
@@ -187,12 +187,13 @@ format_toml_atomic <- function(x,
                                ...,
                                .top_level = FALSE,
                                .formatter) {
-  if (length(x) == 0L) {
+  len <- length(x)
+  if (len == 0L || (len == 1L && isTRUE(is.na(x)))) {
```

Comment:
How about stripping all `NA`s from `x` first:
```r
x <- x[!is.na(x)]
```
Then you just need to test whether the length is 0. And I don't think we'd ever want `NA` to get into a toml file anyways.

Reply:
Great point. I also forgot to update docs...

Reply:
@clauswilke, 
We rely on `dim()` to determine if provided vector should be displayed as an `array`. 
As usual, base-R slicing drops attributes, so we lose information about dimensions. 
We either need to restore dimensions after slicing, or use something like `vctrs::vec_slice()`, we anyway load it indirectly.
What would be the preferred approach?

Reply:
Nvm, I cached dimensions before dropping `NA`s, simple & works fine.